src/witan/app/handler.clj

```diff
@@ -27,24 +31,21 @@
 
 (defn login
   [body]
-  (let [username (:username body)
-        password (:password body)
-        valid? (user/user-valid? username password)]
-    (if valid?
+  (let [{:keys [username password]} body
+        valid-user (user/user-valid? username password)]
+    (if valid-user
       (let [token (user/random-token)]
         (swap! tokens assoc (keyword token) (keyword username))
-        (ok {:token token}))
+        (ok {:token token :id (:id valid-user)}))
       (ok {:message "login failed"}))))
 
 (defn signup
-  [body]
-  (let [username (:username body)
-        password (:password body)]
-    (if (user/add-user! username password)
-      (let [token (user/random-token)]
-        (swap! tokens assoc (keyword token) (keyword username))
-        (created {:token token}))
-      (ok {:message "User already present"}))))
+  [{:keys [username password name] :as body}]
+  (if-let [new-user (user/add-user! body)]
+    (let [token (user/random-token)]
+      (swap! tokens assoc (keyword token) (keyword username))
+      (created {:token token :id (:id new-user)}))
+    (ok {:message "User already present"})))
```

Comment:
Is this a bit leaky in terms of security? That said, not sure what the alternative is without confusing the user.